compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -61,6 +61,7 @@ fn mir_build(tcx: TyCtxt<'_>, def: ty::WithOptConstParam<LocalDefId>) -> Body<'_
         ty::WithOptConstParam { did, const_param_did: None } => {
             tcx.ensure().thir_check_unsafety(did);
             drop(tcx.thir_abstract_const(did));
+            tcx.ensure().check_match(did);
```

Comment:
`check_match` only fetches `thir_body(ty::WithOptConstParam::unknown(did))`, so we only need to ensure if we are stealing this exact one.

Reply:
Makes sense, const args can't contain a match. Is the loop over all body owners in `rustc_interface` still needed then? It seems redundant (except maybe for time passes niceness)

Reply:
AnonConsts can definitely contain patterns to check. If we want to get rid of the `rustc_interface` loop, we'd need to force `check_match` in both cases, with a `WithOptConstParam` as argument. Like we do for `thir_check_unsafety`. I tried to avoid writing all the boilerplate just to remove it once https://github.com/rust-lang/rust/pull/96840 lands.

Reply:
Ah right, makes sense. Seems fine to keep it as-is then